contributing_to_docs/doc_guidelines.adoc

```diff
@@ -863,21 +863,71 @@ Text for admonition
 [id="api-object-formatting"]
 == API object formatting
 
-Use initial capitalization and upper camel case
-(link:https://en.wikipedia.org/wiki/Camel_case#Variations_and_synonyms[PascalCase])
-for Kubernetes/OpenShift API objects and do not mark them up unless referring to
-a specific field or variable name from a spec or manifest.
+For terms that are API objects, the way they are written depends on whether the term is a general reference or an actual reference to the object.
 
-This matches general Kubernetes usage and makes it obvious that a specific
-concept is being referred to. For example:
+[id="api-object-general-references"]
+=== General references
 
-- Pod
-- Deployment
-- Operator
-- DaemonSet (not "daemon set", "daemonset", "daemonSet", or "Daemonset")
+When referring to API objects in general usage, use lowercase and separate multi-word API objects. By general reference, this means any time you are speaking conceptually, or generally, about these components in a cluster. *Default to following this guidance unless you are specifically interacting with/referring to the API object.*
+
+For example:
+
+* pod
+* node
+* daemon set
+* config map
+* deployment
+* deployment config
```

Comment:
This is special, because someone might refer to a `Deployment` object configuration as simply a deployment configuration, as a general term. But that collides with _deployment config_, which is the lowercase version of the `DeploymentConfig` object, which is very different from the `Deployment` object. (I'm not sure _deployment config configuration_ makes sense, either, though.)

Because of this confusion, the deployment sections of the docs need somewhat of a refresh. (OCP also encourages use of `Deployment`, with `DeploymentConfig` eventually be deprecated someday.)

Reply:
@jboxman When I was updating the glossary entries for deployment and deployment config, I was trying to think if we should provide guidance on being more specific in these cases, so as to make the distinction clearer. But I couldn't think of what to say.

I'd be interested to hear if @adellape has any suggestions.

One thing that we could at least do is advise against saying "deployment configuration" to mean a `Deployment` object's configuration.

But it might help to take a look at the sections that we have on deployments and deployment configs, to see how it's going to look and see what guidance we need to provide.

Reply:
I'd say in this case _deployment_ is the concept; And in OCP we happen to, unsurprisingly, have two different ways to accomplish that goal. So we probably need to talk about a deployment generally, and then what each specific `Deployment` or `DeploymentConfig` gets you as an OCP user.

We don't need to belabor `DeploymentConfig`, as from what I've read recently, we'd prefer everyone use upstream `Deployment`. It now supports more or less the functionality of `DeploymentConfig`.

Reply:
That sounds fair to me, though I was wondering if we'd still need to speak of deployment configs generally. But in doing a quick search in the repo, I see a few places where we already are calling them deployments, even though the CLI command is getting a deployment config.

We might need to try it out on a few modules to see how this plays out before we can decide on the guidance.

Reply:
@openshift/team-documentation In case anyone else has any thoughts on this ^

Reply:
+1 to just sticking with to lowercase "deployments" for general references, and use `Deployments` and `DeploymentConfigs` when you need to be specific or distinguish. Also would be interested if/when we can actually sunset DCs from our docs.

Reply:
@jboxman @adellape I updated the glossary to use `Deployment` and `DeploymentConfig`. Can you take a look and see if that makes sense?